### ISSUE TITLE
[Bug] Unable to export documents to pdf

### ISSUE THREAD

**Issue by domanteli0** [cryptee/web-client#50]
Firefox isn't able to export documets to pdfs instead it opens print options.
**To Reproduce**
1. Go to 'Documents Ations'
2. Click export
3. Select pdf
4. Printer menu is now open

**Expected behavior**
1. Go to 'Documents Ations'
2. Click export
3. Select pdf
4. Pdf export menu is open

Here's what happens in chrome:
![gifout](https://user-images.githubusercontent.com/49041398/78066128-e9e17b80-7383-11ea-9949-8e6a85461284.gif)

**Screenshots**
![gifout](https://user-images.githubusercontent.com/49041398/78065357-b5b98b00-7382-11ea-8c7c-b523e3a046ff.gif)

**System Information:**
 - Device: Desktop PC, Phone [Sony F8331]
 - OS: elementaryOS 5.1, Windows 10 and Android 6.0
 - Browser: Windows [Firefox], ElementaryOS [Firefox, Epiphany (webkit-based)] and Chrome on Android
 - Browser Version: Windows [Firefox 74] ElementaryOS [Firefox 74, Epiphany 3.28.6], Android [Chrome 80]
 - Browser Extensions / Add-ons / Content Blockers: ElementaryOS Firefox[Enchanced Tracking Protection, uBlock Origin, Privacy Badger, HTTPS Everywhere, Decentraleyes, Multi-Account Containers (only use it for Facebook, Google, Reddit, Twitter)]; ElementaryOS Epiphany [none]; Windows [Enchanced Tracking Protection only]; Android [none], 
 - Do you have any DNS-based content blockers? No
 - Are you on a managed network? No

**Additional context**
Doesn't happen on the same computer (ElementaryOS) with Chromium (v80.0.3987.87).

I think it's a problem with Firefox and Android. It may be posible to contact Mozilla and Google, although, I'm not sure this is entirely Firefoxs fault. As with Android I think it needs testing on newer devices, maybe it is fixed. Although, again, I'm not sure it's only Androids fault.

**Comment by johnozbay** [comment]
Hi there! 

Many million thanks for the GIFs and thorough report, incredibly helpful! 
And deeply appreciate the time you took to file this bug report! 🙏🏻

First, to provide a little bit of background information on why creating a PDF is a big challenge for a privacy-first productivity service like Cryptee, take a look at my comment here: https://github.com/cryptee/web-client/issues/32#issuecomment-555061881

With what I mentioned in that comment in mind, here's why this is happening:

---

1 – Print behavior isn't consistent across operating systems & browsers 😞
a) On Mac, Firefox, and in general all browsers (can and do) use the system print dialog, which allows for PDF export. Credit where it's due, this is a good move by Apple.

b) On Windows & Linux, this depends on the browser. Chrome and almost all other browsers have Save to PDF in their print dialog. _Except for Firefox_. 🤦🏻‍♂️ For some reason Mozilla didn't build this feature into their print dialog, and require their users to get third party plugins. I have no idea why, but my first guess is that it could even be a licensing issue with Adobe.

2 – You might rightfully say, well at least disable the PDF button on Firefox on Linux & Windows, so that users won't get confused. – I would love to, but the only way to do this would be to check the user agent of the browser to find out the OS, which are incredibly unreliable in 2020. Thanks to malicious websites using the UA to fingerprint users, plugins/extensions and browsers are now making these more and more generic to make it hard to fingerprint users this way. (rightfully so) 

---

Now that I could hopefully paint a complete picture, wearing my solution-hat. 

**Solution 1)** Technically speaking solving #32 will solve this, since once that is ready, we can then generate the PDF in Cryptee, without relying on browser's print dialogs. 

**Solution 2)** In the meantime, I can try making a very hacked-together Cryptee->DOCX->PDF export flow, and theoretically generate a passable PDF, that would definitely lack a lot of styling, but it would still have some stuff like selectable text, fonts, and links etc.

**Solution 3)** I can try disabling this feature on all non-Mac & non-iOS & non-chromium browsers (since statistically, chances are this is a Firefox, and doesn't support print->pdf flow) It's the worst option so far, since even if you have the PDF export plugin, this would hide the export to PDF button, and would be confusing from a UX standpoint. 

---

**TLDR;** 
- Can't send documents to server to generate PDFs (like Google Docs, Office and everyone else does), because of on-device encryption.
- Most third party libraries simply take snapshots of DOM elements to generate PDF, good for QR code tickets etc, but not for documents with links, selectable text etc.
- For the lack of a better option, relying on OS print dialog for PDF exports for now, but it's not guaranteed to have PDF export on Firefox on OSes other than Mac & iOS

I'm all open for suggestions, and would love to make this better for everyone if there's a solution I'm not thinking of. 

Hoping all these make sense! 
I'll keep this issue open until there's a fix to make it easy to track progress & refer to information. ✌🏻

**Comment by domanteli0** [comment]
So turns out It is possible to export to a pdf in Firefox and Android. The problem is that the export dialog looks very similar to a printing one and users might think it is only for printing, while in reality the pdf export option may be hidden behind many other printer options.

![gif](https://user-images.githubusercontent.com/49041398/80454596-d6411a80-8919-11ea-972c-c044ff79b20d.gif)

EDIT:
Well, on Android it is actually easy. I think the dialog has either changed or I just didn't notice.

![Screenshot_2020-04-28_09-32-50](https://user-images.githubusercontent.com/49041398/80455049-90d11d00-891a-11ea-834c-af01ed8d41f7.png)

In my opinion, the best option would be to documents export flows in the helpdesk for users who are lost or have a link to the helpdesk in the export menu.

**Comment by johnozbay** [comment]
Thanks a lot for this! 🙏🏻

I'll write a helpdesk article with screenshots about this and link to the article from the export dialog to help others who might also get stuck some day.

For our internal tracking, if it's good with you as well, I'll go ahead and close this bug✌🏻

Many thanks for this 🙏🏻
Deeply appreciate the help!
